src/components/Modal/src/Modal.vue

```diff
@@ -4,9 +4,47 @@
 			<!-- @slot Modal content -->
 			<slot />
 		</div>
+		<pseudo-window
+			document
+			@keyup.esc="handleEscKey"
+		/>
 	</div>
 </template>
 
+<script>
+import PseudoWindow from 'vue-pseudo-window';
+import modalApi from './modal-api';
+
+export default {
+	components: {
+		PseudoWindow,
+	},
+
+	inject: {
+		modalApi,
+	},
+
+	props: {
+		/**
+		 * Close the modal on ESC
+		 */
+		closeOnEsc: {
+			type: Boolean,
+			default: false,
+		},
```

Comment:
Remove prop. Should be default behavior.

Reply:
We this the prop b/c we don't allow users to close the location modal until location is selected

Reply:
Wondering if this check can be improved. eg. if the Modal has a Cancel/Close button, it automatically supports esc to close, but doesn't if not.

Reply:
Since whether this should be enabled is bound to when the Cancel/Close button passed in, I think it makes more sense to couple the logic too.

Currently, `close-on-esc` needs to be passed into the user-end when the Close button is passed in. It might not be that much more work to instead pass in `<pseudo-window @keyup.esc="modalApi.close()" />` , and move the `hasModalOnTop` check to `close()`.

Taking this thought further, if the esc key is actually triggering the close button, why not add a `trigger-on-esc` prop or support a `@window.esc` event on the button?
```vue
<m-action-bar-button
	...
	@click="modalApi.close()"
	@window-esc="modalApi.close()"
>
	...
</m-action-bar-button>
```

cc @pretzelhammer

Reply:
I decided to add `window-esc` event to the modal and let the developer to close the modal or show alert before closing.

Reply:
The problem I'm surfacing is that the `esc` and clicking Cancel are equivalent, but the logic is spread apart (eg. cancel is handled by button, esc is handled by modal), and therefore you have to be aware of two API surfaces to enable it.

Regarding the before-close confirmation alert, we'll need something more robust to handle cases like the user accidentally closing the window (prompt native alert to prevent), or in the future if we support "clicking out of modal to close" they might accidentally click out of the modal. In Orbit we had a promise based hook `before-close`.

Reply:
@privatenumber I like the `window-esc` prop on the button idea, however how would you expect the case where there are 2 modals stacked on top of each other, both with cancel/close buttons, to work?

Reply:
The `hasModalOnTop` check should move inside modal `close()` to prevent closing modals that aren't active.

This should actually already be there because the modals below are only hidden via `opacity: 0`, so there is currently a bug where you can actually "tab" back and "click" on the close button of a hidden modal. As a side note, have you considered doing `display: none` on it after the fade-out? This will be an accessibility issue for those navigating with a keyboard.

On Slack, @sohnj877 also wondered if there's a case where we want the "esc to close" but no Cancel button. I don't think that should ever happen either because it's inaccessible to those that can't use a keyboard.